packages/data-fetcher/src/transaction/transactionTraces.service.ts

```diff
@@ -34,27 +36,54 @@ export interface TransactionTraceData {
 interface ExtractedTraceData {
   contractAddresses: ContractAddress[];
   transfersWithValue: Transfer[];
+  internalTransactions: InternalTransaction[];
   error?: string;
   revertReason?: string;
+  traceIndex: number;
 }
 
 function getTransactionTraceData(
   block: Block,
   transaction: TransactionResponse,
   transactionTrace: TransactionTrace,
-  extractedData: ExtractedTraceData = undefined
+  extractedData: ExtractedTraceData = undefined,
+  traceAddress = ""
 ): ExtractedTraceData {
   if (!extractedData) {
     extractedData = {
       contractAddresses: [],
       transfersWithValue: [],
+      internalTransactions: [],
       error: transactionTrace?.error,
       revertReason: transactionTrace?.revertReason,
+      traceIndex: 0,
     };
   }
 
   if (transactionTrace) {
     const traceType = transactionTrace.type.toLowerCase();
+    const currentTraceAddress = traceAddress || "0";
+    const currentTraceIndex = extractedData.traceIndex++;
+    const internalTx: InternalTransaction = {
+      transactionHash: transaction.hash,
+      blockNumber: transaction.blockNumber,
+      from: transactionTrace.from.toLowerCase(),
+      to: transactionTrace.to?.toLowerCase(),
+      value: BigInt(transactionTrace.value || "0x0").toString(),
+      gas: transactionTrace.gas ? BigInt(transactionTrace.gas).toString() : undefined,
+      gasUsed: transactionTrace.gasUsed ? BigInt(transactionTrace.gasUsed).toString() : undefined,
+      input: transactionTrace.input || "0x",
+      output: transactionTrace.output || "0x",
+      type: traceType.toUpperCase(),
```

Comment:
what's the point of having `type` and `callType` and storing in the DB if the values are basically the same?